SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp

```diff
@@ -157,10 +161,67 @@ void USpatialReceiver::OnRemoveEntity(Worker_RemoveEntityOp& Op)
 
 void USpatialReceiver::OnAuthorityChange(Worker_AuthorityChangeOp& Op)
 {
-	if (Op.component_id == SpatialConstants::GLOBAL_STATE_MANAGER_COMPONENT_ID
-		&& Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	if (bInCriticalSection)
+	{
+		PendingAuthorityChanges.Add(Op);
+		return;
+	}
+
+	HandleActorAuthority(Op);
+}
+
+void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
+{
+	if (NetDriver->IsServer())
+	{
+		if (Op.component_id == SpatialConstants::GLOBAL_STATE_MANAGER_COMPONENT_ID
+			&& Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+		{
+			GlobalStateManager->ExecuteInitialSingletonActorReplication();
+			return;
+		}
+
+		if (Op.component_id == SpatialConstants::POSITION_COMPONENT_ID)
+		{
+			if (AActor* Actor = NetDriver->GetEntityRegistry()->GetActorFromEntityId(Op.entity_id))
+			{
+				if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+				{
+					Actor->Role = ROLE_Authority;
+					if (Actor->GetNetConnection() != nullptr)
+					{
+						Actor->RemoteRole = ROLE_AutonomousProxy;
+					}
+					// TODO: Change to see it has an APlayerController
```

Comment:
Add Jira ticket if you don't intend to fix this with the PR

Reply:
I fixed it now rather than adding a ticket.